src/mpid/ch4/src/ch4_recv.h

```diff
@@ -136,19 +139,32 @@ MPL_STATIC_INLINE_PREFIX int MPID_Recv_init(void *buf,
             MPIR_ERR_POP(mpi_errno);
         }
 
-        mpi_errno =
-            MPIDI_NM_mpi_recv_init(buf, count, datatype, rank, tag, comm, context_offset, av,
-                                   &(MPIDI_CH4I_REQUEST_ANYSOURCE_PARTNER(*request)));
+        MPIDI_CH4I_REQUEST(*request, is_local) = 1;
 
-        if (mpi_errno != MPI_SUCCESS) {
-            MPIR_ERR_POP(mpi_errno);
-        }
+        if (!MPIR_Request_is_complete(*request) && !MPIDI_CH4U_REQUEST_IN_PROGRESS(*request)) {
```

Comment:
I have two concerns in this line:
1. This might not be quite correct from the patch ordering point of view -- MPIDI_CH4U_REQUEST_IN_PROGRESS assumes that the shmmod uses the active message code but it does not at this moment.
2. Also, is checking for MPIR_Request_is_complete(*request) valid here? Here *request is an inactive persistent request, which is always considered as "completed" if I understand correctly. So, this condition will completely disable NM path.

Reply:
Oops. I copied your comments over as well. 😄 See my responses below.